### DISCUSSION TITLE
Is RDF4J affected by CVE-2022-42889: Apache Commons Text prior to 1.10.0 allows RCE when applied to untrusted input due to insecure interpolation defaults

### DISCUSSION THREAD

**Discussion by PravinPrameKumar** [eclipse-rdf4j/rdf4j#4318] [Q&A] [answered]
Does RDF4J internally use any string interpolation which uses input from user queries? If so, is there a chance that this CVE applies?

   Apache Commons Text jars within the vulnerable version range found: 
* \RDFStore\apache-tomcat\webapps\rdf4j-server\WEB-INF\lib\commons-text-1.9.jar 
*\RDFStore\apache-tomcat\webapps\rdf4j-workbench\WEB-INF\lib\commons-text-1.9.jar 
* \RDFStore\Eclipse RDF4J SDK\lib\commons-text-1.9.jar

CVE-2022-42889: Apache Commons Text prior to 1.10.0 allows RCE when applied to untrusted input due to insecure interpolation defaults 
Severity: important
Description:
Apache Commons Text performs variable interpolation, allowing properties to be dynamically evaluated and expanded. The standard format for interpolation is "${prefix:name}", where "prefix" is used to locate an instance of org.apache.commons.text.lookup.StringLookup that performs the interpolation. Starting with version 1.5 and continuing through 1.9, the set of default Lookup instances included interpolators that could result in arbitrary code execution or contact with remote servers. These lookups are: - "script" - execute expressions using the JVM script execution engine (javax.script) - "dns" - resolve dns records - "url" - load values from urls, including from remote servers Applications using the interpolation defaults in the affected versions may be vulnerable to remote code execution or unintentional contact with remote servers if untrusted configuration values are used. Users are recommended to upgrade to Apache Commons Text 1.10.0, which disables the problematic interpolators by default.

**Comment by abrokenjester** [comment, marked as answer]
Since release 4.2.1, RDF4J is using commons-text 1.10.0 (see GH-4233). So unless you use a version of RDF4J older than that, it's not affected by this CVE.

However even for older versions of RDF4J this is unlikely to be an issue, since (as far as I'm aware) we do not make use variable interpolation anywhere in the project.

**Reply by PravinPrameKumar** [reply]
Thank you for your prompt help as always, Jeen. :)
